aea/protocols/default/message.py

```diff
@@ -59,20 +59,47 @@ def __init__(self, type: Optional[Type] = None,
         super().__init__(type=type, **kwargs)
         assert self.check_consistency(), "DefaultMessage initialization inconsistent."
 
+    @property
+    def type(self) -> Type:  # noqa: F821
```

Comment:
@DavidMinarsch what about this? It is needed because `Type` is defined only when the `DefaultMessage` is defined (since `Type` is an inner class), but that is not true at this point for `flake8`.

Is it OK? Or should we move the inner class outside (but that might be a potentially breaking change)?

Reply:
Yeah I don;t like the `# noqa: F821`...though I thought there was a way to hint the type in such a scenario. I will have a look ... Definitely would not move it out.